services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/oauth/OAuthRefreshTokenManagerImplUnitTest.java

```diff
@@ -0,0 +1,514 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.platform.commons.util.StringUtils;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.sagebionetworks.repo.model.AuthorizationConstants;
+import org.sagebionetworks.repo.model.ConflictingUpdateException;
+import org.sagebionetworks.repo.model.UnauthorizedException;
+import org.sagebionetworks.repo.model.UserInfo;
+import org.sagebionetworks.repo.model.auth.OAuthClientDao;
+import org.sagebionetworks.repo.model.auth.OAuthRefreshTokenDao;
+import org.sagebionetworks.repo.model.oauth.OAuthClientAuthorizationHistoryList;
+import org.sagebionetworks.repo.model.oauth.OAuthRefreshTokenInformation;
+import org.sagebionetworks.repo.model.oauth.OAuthRefreshTokenInformationList;
+import org.sagebionetworks.repo.model.oauth.OAuthScope;
+import org.sagebionetworks.repo.model.oauth.OIDCClaimsRequest;
+import org.sagebionetworks.repo.web.NotFoundException;
+import org.sagebionetworks.util.Clock;
+
+@ExtendWith(MockitoExtension.class)
+public class OAuthRefreshTokenManagerImplUnitTest {
+
+	@Mock
+	OAuthClientDao mockOAuthClientDao;
+
+	@Mock
+	OAuthRefreshTokenDao mockOAuthRefreshTokenDao;
+
+	@Mock
+	Clock clock;
+
+	@InjectMocks
+	OAuthRefreshTokenManagerImpl oauthRefreshTokenManager;
+
+	private static final Long EXPECTED_LEASE_DURATION = 180L;
+	private static final Long EXPECTED_MAX_REFRESH_TOKENS = 100L;
+
+	// Some constants to simplify test setup
+	private static final String TOKEN_ID = "1234";
+	private static final String USER_ID = "4321";
+	private static final String CLIENT_ID = "67890";
+
+	private static final UserInfo USER_INFO = new UserInfo(false, Long.valueOf(USER_ID));
+
+	@Test
+	public void testCreateRefreshToken() {
+		List<OAuthScope> scopes = Arrays.asList(OAuthScope.authorize, OAuthScope.openid);
+		OIDCClaimsRequest claimsRequest = new OIDCClaimsRequest();
+		claimsRequest.setUserinfo(Collections.emptyMap());
+		claimsRequest.setId_token(Collections.emptyMap());
+
+		when(clock.now()).thenReturn(new Date()); // Ensure all 'clock.now()' calls return the same time
+		when(mockOAuthRefreshTokenDao.createRefreshToken(anyString(), any(OAuthRefreshTokenInformation.class)))
```

Comment:
To make consistent with other tests:
ArgumentCaptor< OAuthRefreshTokenInformation> captor = ArgumentCaptor.forClass(OAuthRefreshTokenInformation.class);
when(mockOAuthRefreshTokenDao.createRefreshToken(anyString(), captor.capture())).thenReturn(*whatever*);
// ... call method under test
OAuthRefreshTokenInformation captured = captor.Value();
assertTrue(StringUtils.isNotBlank(captured.getName()));